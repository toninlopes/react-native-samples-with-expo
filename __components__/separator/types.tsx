import { ColorValue } from "react-native";

export type SeparatorProps = {
  color?: ColorValue | undefined;
  style?: 'solid' | 'dotted' | 'dashed' | undefined;
};

export type SeparatorType = React.FC<SeparatorProps>;