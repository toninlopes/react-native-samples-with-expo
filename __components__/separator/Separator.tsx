import React from "react";
import { ColorValue, StyleSheet, View } from "react-native";
import type { SeparatorProps } from './types';

const Separator: React.FC<SeparatorProps> = ({ color, style }) => {
  return (
    <View style={[styles().separator, styles(color, style).border]} />
  );
};

const styles = (
  color?: ColorValue | undefined,
  style?: 'solid' | 'dotted' | 'dashed' | undefined,
) => ({
  ...StyleSheet.create({
    separator: {
      flex: 1,
    },
    border: {
      borderWidth: 0.8,
      borderColor: color || 'silver',
      borderStyle: style || 'dashed',
    },
  }),
});

export default Separator;