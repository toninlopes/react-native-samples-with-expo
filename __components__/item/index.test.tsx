import React from 'react';
import renderer from 'react-test-renderer';
import { Item } from '.';
import { Separator } from '../separator';

describe('Item', () => {
  test('When there is none child.', () => {
    const item = renderer.create(<Item />).toJSON();
    expect(item).toMatchSnapshot();
  });

  test('When there is row as child.', () => {
    const item = renderer.create((
      <Item>
        <Item.Row>
          <Item.Content title='Title' />
        </Item.Row>
      </Item>
    )).toJSON();
    expect(item).toMatchSnapshot();
  });

  test('When there are row, title, chevron and separator.', () => {
    const item = renderer.create((
      <Item>
        <Item.Row>
          <Item.Content title='Title' />
          <Item.Chevron />
        </Item.Row>
        <Separator style='dashed'/>
      </Item>
    )).toJSON();
    expect(item).toMatchSnapshot();
  });

  test('When there are row and separator.', () => {
    const item = renderer.create((
      <Item>
        <Item.Row>
          <Item.Content title='Title' />
        </Item.Row>
        <Separator style='dashed'/>
      </Item>
    )).toJSON();
    expect(item).toMatchSnapshot();
  });
});