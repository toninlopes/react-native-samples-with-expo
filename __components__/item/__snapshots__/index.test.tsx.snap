// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item When there are row and separator. 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingVertical": 20,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 22,
          }
        }
      >
        Title
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "flex": 1,
        },
        {
          "borderColor": "silver",
          "borderStyle": "dashed",
          "borderWidth": 0.8,
        },
      ]
    }
  />
</View>
`;

exports[`Item When there are row, title, chevron and separator. 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingVertical": 20,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 22,
          }
        }
      >
        Title
      </Text>
    </View>
    <View
      style={{}}
    >
      <Text />
    </View>
  </View>
  <View
    style={
      [
        {
          "flex": 1,
        },
        {
          "borderColor": "silver",
          "borderStyle": "dashed",
          "borderWidth": 0.8,
        },
      ]
    }
  />
</View>
`;

exports[`Item When there is none child. 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
/>
`;

exports[`Item When there is row as child. 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingVertical": 20,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 22,
          }
        }
      >
        Title
      </Text>
    </View>
  </View>
</View>
`;
